src/hotspot/share/code/codeBlob.inline.hpp

```diff
@@ -33,7 +33,7 @@
 
 inline const ImmutableOopMap* CodeBlob::oop_map_for_slot(int slot, address return_address) const {
   assert(_oop_maps != nullptr, "nope");
-  return _oop_maps->find_map_at_slot(slot, (intptr_t) return_address - (intptr_t) code_begin());
+  return _oop_maps->find_map_at_slot(slot, delta_as_int((intptr_t) return_address, (intptr_t) code_begin()));
```

Comment:
Hm.  Maybe this would be ok. Our original idea was to make it T* not T until this cast.  I don't know how many other cases there are that I haven't gotten to yet.  But it would eliminate a cast, so that's good (unless these aren't the same).  Some instances have ptr - constant that gets promoted I think.
The reason we didn't pick pointer_delta_as_int because pointer_delta has different semantics.  pointer_delta insists on positive results.

Reply:
Taking out that cast does work, so I've fixed that.

Reply:
> pointer_delta has different semantics

Right. That was "recently" added to pointer_delta with JDK-8260046. It begs the question why felt the need to add it there but feel that it is OK to skip it for delta_as_int? Is there some usage of delta_as_int that gives back negative values? Could that call site be changed?

Reply:
There are sites where the result is negative but this is a good suggestion because it makes the name more consistent.  I can change those to plain check_cast<int>s.